test/stmt/switch_stmt1.swift

```diff
@@ -5,9 +5,35 @@ enum E {
   case e2
 }
 
+enum E2 {
+  case `init`, `subscript`, foo, bar
+}
+
 func foo1(e : E, i : Int) {
   switch e {} // expected-error{{switch must be exhaustive}}
   // expected-note@-1 {{missing case: '.e1'}}
   // expected-note@-2 {{missing case: '.e2'}}
   switch i {} // expected-error{{'switch' statement body must have at least one 'case' or 'default' block; do you want to add a default case?}}{{13-13=default:\n<#code#>\n}}
 }
+
+func foo2(e: E2) {
+  switch (e) {
+  case .init: break
+  // expected-error@-1 {{keyword 'init' cannot be used as an identifier here}}
```

Comment:
Should it be a warning? Are there really no valid reasons to use it?

Reply:
I think the intent is that if there were one, we wouldn't have an `UnresolvedMemberExpr` here, but I can't think of one without the parens. We should probably have a test case for when you *do* have parens though, for ``case `init`(Int)`` in the enum.

Reply:
To be clear, I think it's okay to not have a special diagnostic for that right now. We just need to Not Crash.

Reply:
Is it actually crashing? I thought it was a non-exhaustiveness issue.

Reply:
It doesn’t crash but rather produces ‘add missing case: .init’ diagnostic currently.

Reply:
No, I don't think it's crashing now, sorry. I want it to continue not crashing, though.

Reply:
See my comment below.